agent/src/com/cloud/agent/AgentShell.java

```diff
@@ -107,18 +111,48 @@ public String getPod() {
     }
 
     @Override
-    public String getHost() {
-        final String[] hosts = _host.split(",");
+    public String getNextHost() {
+        final String[] hosts = getHosts();
         if (_hostCounter >= hosts.length) {
             _hostCounter = 0;
         }
-        final String host = hosts[_hostCounter % hosts.length];
+        connectedHost = hosts[_hostCounter % hosts.length];
         _hostCounter++;
-        return host;
+        s_logger.info("Connecting to host: " + connectedHost);
```

Comment:
This method only returns the next possible host in list, however, this does not mean we're connected to the host yet. This might cause threading issues, and in rare cases the connectedHost might not be the connected host. Reconsider how this could be re-written, perhaps the connectedHost can be updated using one of the `Task.CONNECT` handlers.

Reply:
Done, thanks! Please checkout `hostToConnect` and `connectedHost` relationship using `updateConnectedHost()` method on connection to a host.